.github/workflows/release.yml

```diff
@@ -0,0 +1,59 @@
+
+name: Version Release Check
+
+on:
+  release:
+    types: [published]
+  workflow_call:
+    inputs:
+      release-tag:
+        required: true
+        type: string
+
+env:
+  RUST_BACKTRACE: 1
+  SHELL: /bin/bash
+  CARGO_INCREMENTAL: 0
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - id: tag-name
+        run: |
+          if [[ '${{ inputs.release-tag }}' != '' ]]; then
+            echo "RELEASE_TAG=${{ inputs.release-tag }}" >> ${GITHUB_OUTPUT}
+          elif [[ '${{ github.event_name }}' == 'release' ]]; then
+            echo "RELEASE_TAG=${{ github.ref_name }}" >> ${GITHUB_OUTPUT}
+          fi
+    outputs:
+      tag: ${{ steps.tag-name.outputs.RELEASE_TAG }}
```

Comment:
I think we could rewrite this as env var: `RELEASE_TAG: ${{ github.event_name == 'release' && github.ref_name ||  inputs.release-tag }}`.

Reply:
Thanks for the suggestion. I didn't know github action variable can simplify to this.